bika/lims/utils/__init__.py

```diff
@@ -890,6 +891,54 @@ def is_bika_installed():
     qi = api.portal.get_tool("portal_quickinstaller")
     return qi.isProductInstalled("bika.lims")
 
+def convert_unit(result, formula, precision):
+    """ take a value and a unit conversion formula and convert the result
+    """
+    try:
+        formula = formula.replace('Value', '%f')
+        new =  eval(formula % float(result))
```

Comment:
I would reduce here the set of __builtins__ (see Calculation module) to make the eval not so evil as it might be

Reply:
I made an attempt @ramonski , is it now correct